pandas/core/arrays/base.py

```diff
@@ -610,3 +618,121 @@ def _ndarray_values(self):
         used for interacting with our indexers.
         """
         return np.array(self)
+
+
+class ExtensionOpsBase(object):
+    """
+    A base class for the mixins for different operators.
+    Can also be used to define an individual method for a specific
+    operator using the class method create_method()
+    """
+    @classmethod
+    def create_method(cls, op):
+        """
+        A class method that returns a method that will correspond to an
+        operator for an ExtensionArray subclass.
+
+        Parameters
+        ----------
+        op:   An operator that takes arguments op(a, b)
+
+        Returns
+        -------
+        A method that can be bound to a method of a class
+
+        Usage
+        -----
+        Given an ExtensionArray subclass called MyClass, use
+
+        mymethod = create_method(my_operator)
+        in the class definition of MyClass to create the operator
+
+        """
+        op_name = ops._get_op_name(op, False)
+
+        def _binop(self, other):
+            def convert_values(parm):
+                if isinstance(parm, ExtensionArray):
+                    ovalues = list(parm)
+                elif is_extension_array_dtype(parm):
+                    ovalues = parm.values
+                elif is_list_like(parm):
+                    ovalues = parm
+                else:  # Assume its an object
+                    ovalues = [parm] * len(self)
+                return ovalues
+            lvalues = convert_values(self)
+            rvalues = convert_values(other)
+
+            try:
+                res = [op(a, b) for (a, b) in zip(lvalues, rvalues)]
+            except TypeError:
+                msg = ("ExtensionDtype invalid operation " +
+                       "{opn} between {one} and {two}")
+                raise TypeError(msg.format(opn=op_name,
+                                           one=type(lvalues),
+                                           two=type(rvalues)))
+
+            res_values = com._values_from_object(res)
+
+            try:
+                res_values = self._from_sequence(res_values)
+            except TypeError:
+                pass
+
+            return res_values
+
+        name = '__{name}__'.format(name=op_name)
+        return set_function_name(_binop, name, cls)
+
+
+class ExtensionArithmeticMixin(ExtensionOpsBase):
```

Comment:
inherit from object

Reply:
You will need to be clearer what your objection is. In the way @Dr-Irv implemented it, he needs to subclass the other class, to have the `create_method` implementation. If you don't like the way how it is currently split in three classes, be specific about that.

Reply:
same point as above ``create_method`` needs to return NotImplementedError in this class, hence you don't need a base class. I am being pedantic on purpose here. What you are doing is coupling way to tight the scalar implementation.

Reply:
@jreback can't inherit from object, as the `set_function_name` call in `create_method()` needs to know the parent class.  (or maybe you know some other way to do this)

I don't see the point that `create_method` needs to return NotImplementedError, as the whole point of what I'm doing here is to just provide the capability to have the operators on `ExtensionArray` use the underlying operators of the elements on the array.